src/main/java/org/wildfly/security/auth/server/ModifiableRealmIdentity.java

```diff
@@ -47,30 +47,13 @@
     void create() throws RealmUnavailableException;
 
     /**
-     * Replace map of credentials of this identity.  If the identity does not exist, an exception is thrown.
+     * Set the credentials of this identity.  If the identity does not exist, an exception is thrown.
+     * Any existing credential(s) are replaced/updated with the new value (in a possibly realm-specific manner).
```

Comment:
We could possibly change the contract to accept a credential type, since our credential types fall under broad categories (password, public key, certificate, etc.).  I agree with your later comment that we should do this in a follow-up though.